tests/kafkatest/services/kafka/templates/kafka.properties

```diff
@@ -23,6 +23,16 @@ listener.security.protocol.map={{ listener_security_protocol_map }}
 
 inter.broker.listener.name={{ interbroker_listener.name }}
 
+{% for k, v in client_listener_overrides.iteritems() %}
+listener.name.{{ security_protocol.lower() }}.{{ security_config.client_sasl_mechanism.lower() }}.{{ k }}={{ v }}
```

Comment:
The name `client_listener_overrides` implies these could be any listener override (e.g. SSL keystore), but this config entry assumes they are only SASL configs with the SASL mechanism prefix.

Reply:
@brianbushree Listener configs may be:
```
listener.name.replication.ssl.keystore.location=/mnt/keystore.jks
```
OR
```
listener.name.external.oauthbearer.sasl.jaas.config=...
```
The first should not have SASL mechanism in the config name, but the second requires it. We will need to support both at some point. In this PR, it is sufficient to support SASL ones, but we need to make sure it doesn't prevent SSL configs from being added later without changing tests. We could do one of these:
1) Separate out the configs into SASL configs which are mechanism prefixed and other listener configs that are not. Caller will need to configure these correctly based on the config being overridden. For example, we could rename `ListenerConfig.client_listener_overrides` to `ListenerConfig.client_sasl_listener_overrides` and `interbroker_listener_overrides` to `interbroker_sasl_listener_overrides`.
2) Expect the caller to pass in `oauthbearer.sasl.jaas.config` for the second case
3) Work out whether SASL is being used for the listener and add mechanism prefix if required. Caller doesn't need any changes if we can do this automatically.